File: LeetCode/557-Reverse-Words-in-a-String-III.cpp
class Solution {
public:
    string reverseWords(string s) {
        string ans="", tmp="";
        for(auto letter: s) {
            if(letter==' ') {
                reverse(tmp.begin(), tmp.end());
                ans+=(tmp+' ');
                tmp="";
                continue;
            }
            tmp+=letter;
        }
        reverse(tmp.begin(), tmp.end());
        ans+=tmp;
        return ans;
    }
};

File: LeetCode/557-Reverse-Words-in-a-String-III.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s=list(s)
        l=0
        for r in range(len(s)):
            if s[r]==" " or r==len(s)-1:
                tl, tr=l, r if r==len(s)-1 else r-1

                while tl<tr:
                    s[tl], s[tr]=s[tr], s[tl]
                    tl+=1
                    tr-=1
                l=r+1

        return "".join(s)